Tests/SubprocessTests/IntegrationTests.swift

```diff
@@ -0,0 +1,2296 @@
+            output == message
+        )
+    }
+
+    // Windows does not support argument 0 override
+    // This test will not compile on Windows
+    @Test func testArgumentsOverride() async throws {
+        #if os(Windows)
+        let setup = TestSetup(
+            executable: .name("powershell.exe"),
+            arguments: .init(
+                executablePathOverride: "apple",
+                remainingValues: ["-Command", "[Environment]::GetCommandLineArgs()[0]"]
+            )
+        )
+        #else
+        let setup = TestSetup(
+            executable: .path("/bin/sh"),
+            arguments: .init(
+                executablePathOverride: "apple",
+                remainingValues: ["-c", "echo $0"]
+            )
+        )
+        #endif
+        let result = try await _run(
+            setup,
+            input: .none,
+            output: .string(limit: 16),
+            error: .discarded
+        )
+        #expect(result.terminationStatus.isSuccess)
+        // rdar://138670128
+        // https://github.com/swiftlang/swift/issues/77235
+        let output = result.standardOutput?
+            .trimmingNewLineAndQuotes()
+        #expect(
+            output == "apple"
+        )
+    }
+
+    #if !os(Windows)
+    // Windows does not support byte array arguments
+    // This test will not compile on Windows
+    @Test func testArgumentsFromBytes() async throws {
+        let arguments: [UInt8] = Array("Data Content\0".utf8)
+        let result = try await Subprocess.run(
+            .path("/bin/echo"),
+            arguments: .init(
+                executablePathOverride: nil,
+                remainingValues: [arguments]
+            ),
+            output: .string(limit: 32)
+        )
+        #expect(result.terminationStatus.isSuccess)
+        // rdar://138670128
+        // https://github.com/swiftlang/swift/issues/77235
+        let output = result.standardOutput?
+            .trimmingCharacters(in: .whitespacesAndNewlines)
+        #expect(
+            output == "Data Content"
+        )
+    }
+    #endif
+}
+
+// MARK: - Environment Tests
+extension SubprocessIntegrationTests {
+    @Test func testEnvironmentInherit() async throws {
+        #if os(Windows)
+        let setup = TestSetup(
+            executable: .name("cmd.exe"),
+            arguments: ["/c", "echo %Path%"],
+            environment: .inherit
+        )
+        #else
+        let setup = TestSetup(
+            executable: .path("/bin/sh"),
+            arguments: ["-c", "printenv PATH"],
+            environment: .inherit
+        )
+        #endif
+        let result = try await _run(
+            setup,
+            input: .none,
+            output: .string(limit: .max),
+            error: .discarded
+        )
+        #expect(result.terminationStatus.isSuccess)
+        let pathValue = try #require(result.standardOutput)
+        #if os(Windows)
+        let expected = try #require(ProcessInfo.processInfo.environment["Path"])
```

Comment:
Path/PATH must be treated case-insensitively as there's no guarantee of its capitalization form.